x/evm/precompile/funtoken_test.go

```diff
@@ -428,3 +428,226 @@ func (s *FuntokenSuite) TestPrecompileLocalGas() {
+	return new(big.Int).Mul(big.NewInt(n), e18)
+}
+
+func (s *FuntokenSuite) TestSendToEvm_NotMadeFromCoin() {
+	deps := evmtest.NewTestDeps()
+
+	bob := evmtest.NewEthPrivAcc()
+	alice := evmtest.NewEthPrivAcc()
+
+	// Fund user so they can create funtoken from an ERC20
+	createFunTokenFee := deps.EvmKeeper.FeeForCreateFunToken(deps.Ctx)
+	s.Require().NoError(testapp.FundAccount(
+		deps.App.BankKeeper, deps.Ctx, deps.Sender.NibiruAddr,
+		createFunTokenFee,
+	))
+
+	// Deploy an ERC20 with 18 decimals
+	erc20Resp, err := evmtest.DeployContract(&deps, embeds.SmartContract_TestERC20)
+	s.Require().NoError(err, "failed to deploy test ERC20")
+	erc20Addr := erc20Resp.ContractAddr
+
+	// the initial supply was sent to the deployer
+	evmtest.AssertERC20BalanceEqual(s.T(), deps, erc20Addr, deps.Sender.EthAddr, bigTokens(1000000))
+
+	// create fun token from that erc20
+	_, err = deps.EvmKeeper.CreateFunToken(sdk.WrapSDKContext(deps.Ctx), &evm.MsgCreateFunToken{
+		Sender:    deps.Sender.NibiruAddr.String(),
+		FromErc20: &eth.EIP55Addr{Address: erc20Addr},
+	})
+	s.Require().NoError(err)
+
+	// Transfer 500 tokens to bob => 500 * 10^18 raw
+	deployerAddr := gethcommon.HexToAddress(erc20Resp.EthTxMsg.From)
+	_, err = deps.EvmKeeper.CallContract(
+		deps.Ctx,
+		embeds.SmartContract_TestERC20.ABI,
+		deployerAddr,
+		&erc20Addr,
+		true,
+		keeper.Erc20GasLimitExecute,
+		"transfer",
+		bob.EthAddr,
+		bigTokens(500), // 500 in human sense
+	)
+	s.Require().NoError(err)
+
+	// Now user should have 500 tokens => raw is 500 * 10^18
+	evmtest.AssertERC20BalanceEqual(s.T(), deps, erc20Addr, bob.EthAddr, bigTokens(500))
+
+	// sendToBank: e.g. 100 tokens => 100 * 1e18 raw
+	input, err := embeds.SmartContract_FunToken.ABI.Pack(
+		string(precompile.FunTokenMethod_sendToBank),
+		[]any{
+			erc20Addr, // address
+			bigTokens(100),
+			alice.NibiruAddr.String(),
+		}...,
+	)
+	s.Require().NoError(err)
+	_, resp, err := evmtest.CallContractTx(&deps, precompile.PrecompileAddr_FunToken, input, bob)
+	s.Require().NoError(err)
+	s.Require().Empty(resp.VmError)
+
+	// Bank side should see 100
+	bankBal := deps.App.BankKeeper.GetBalance(deps.Ctx, alice.NibiruAddr, "erc20/"+erc20Addr.Hex())
+	s.Require().EqualValues(bigTokens(100), bankBal.Amount.BigInt())
+
+	// Expect user to have 400 tokens => 400 * 10^18
+	evmtest.AssertERC20BalanceEqual(s.T(), deps, erc20Addr, bob.EthAddr, bigTokens(400))
+
+	// Finally sendToEvm(100)
+	input2, err := embeds.SmartContract_FunToken.ABI.Pack(
+		"sendToEvm",
+		[]any{
+			bankBal.Denom,
+			bigTokens(100),
+			bob.EthAddr.Hex(),
+		}...,
+	)
+	s.Require().NoError(err)
+	_, resp2, err := evmtest.CallContractTx(&deps, precompile.PrecompileAddr_FunToken, input2, alice)
+	s.Require().NoError(err)
+	s.Require().Empty(resp2.VmError)
+
+	// no bank side left
+	balAfter := deps.App.BankKeeper.GetBalance(deps.Ctx, alice.NibiruAddr, bankBal.Denom).Amount.BigInt()
+	s.Require().EqualValues(bigTokens(0), balAfter)
+
```

Comment:
Can you also check the EVM module account's balance? It should be zero as well here.